### ISSUE TITLE
NRE while build

### ISSUE THREAD

**Issue by mikasoukhov** [EWSoftware/SHFB#550]
After VS 2017 update got the issue

> MRefBuilder : error : An unexpected error occurred trying to execute the MRefBuilder MSBuild task: System.NullReferenceException: Object reference not set to an instance of an object. [C:\StockSharp\StockSharp (GitHub)\Documentation\ru\Help\Working\GenerateRefInfo.proj]
> MRefBuilder : error :    at Microsoft.Ddue.Tools.Reflection.ApiFilter.IsExposedType(TypeNode type) [C:\StockSharp\StockSharp (GitHub)\Documentation\ru\Help\Working\GenerateRefInfo.proj]
> MRefBuilder : error :    at Microsoft.Ddue.Tools.Reflection.ApiFilter.NamespaceContainsExposedTypes(Namespace space) [C:\StockSharp\StockSharp (GitHub)\Documentation\ru\Help\Working\GenerateRefInfo.proj]
> MRefBuilder : error :    at Microsoft.Ddue.Tools.Reflection.ApiFilter.IsExposedNamespace(Namespace space) [C:\StockSharp\StockSharp (GitHub)\Documentation\ru\Help\Working\GenerateRefInfo.proj]
> MRefBuilder : error :    at Microsoft.Ddue.Tools.Reflection.ApiVisitor.VisitApis() [C:\StockSharp\StockSharp (GitHub)\Documentation\ru\Help\Working\GenerateRefInfo.proj]
> MRefBuilder : error :    at Microsoft.Ddue.Tools.MRefBuilderCore.Main(String[] args) [C:\StockSharp\StockSharp (GitHub)\Documentation\ru\Help\Working\GenerateRefInfo.proj]
> MRefBuilder : error :    at Microsoft.Ddue.Tools.MSBuild.MRefBuilder.Execute() [C:\StockSharp\StockSharp (GitHub)\Documentation\ru\Help\Working\GenerateRefInfo.proj]
> Done Building Project "C:\StockSharp\StockSharp (GitHub)\Documentation\ru\Help\Working\GenerateRefInfo.proj" (default targets) -- FAILED.

**Comment by EWSoftware** [comment]
Can you attach or send me an example that I can use to duplicate the problem?

**Comment by mikasoukhov** [comment]
Please unzip this package https://yadi.sk/d/Sh9IQQVy3Qdi7k and try to compile \SHFB_Issue_550\StockSharp (GitHub)\Documentation\ru\Documentation.shfbproj

**Comment by mikasoukhov** [comment]
Also it has a millions warning related with 3-th party libs like DevExpress. Could to please assist how to suppress it and see only related warnings?

**Comment by EWSoftware** [comment]
You can use the Additional Reference Links plug-in to suppress warnings about unresolved reference links in third party reference assemblies.

**Comment by mikasoukhov** [comment]
Maybe the issue related with that one https://github.com/EWSoftware/SHFB/issues/552 

Is the SHFB is not compatible with VS 2017?

**Comment by EWSoftware** [comment]
No, #552 is unrelated.  That only applies to the standalone GUI and is resolved as noted by installing the MSBuild Tools 2015.

**Comment by EWSoftware** [comment]
It looks like some of your assemblies have obfuscated code in them.  Some of the metadata for those types is null which is causing MRefBuilder to fail.  I've added null checks at the points where it's getting the null references on the obfuscated types.  It'll be a couple of weeks before I get a new release published.

**Comment by mikasoukhov** [comment]
Please send me preview patch and I will try to test it. Also I know here is
some problems with Additional refs (last time it causes UI crash but I
cannot get the behavior cause the build stopped on first stages). It would
great to fix it as well.
-- 
Mikhail Sukhov,
StockSharp LLC, http://stocksharp.com

**Comment by EWSoftware** [comment]
I've made some changes to the core assemblies to move some stuff around and some new project properties have been added so the SHFB projects will need upgrading.  As such, I can't just build a patched version of MRefBuilder.exe.  You'll need to download and build the code for all the tools if you want to try the fix out before the next release.  The latest changes containing the fix have been checked in.
